### ISSUE TITLE
Cannot export builders in TS projects that generate declaration

### ISSUE THREAD

**Issue by crohn** [jackfranklin/test-data-bot#361]
In my project `tsconfig.json` has `declaration` flag set.

```json
{
  "compilerOptions": {
    "declaration": true
  }
}
```

I want to create and export a builder for reuse.

```typescript
import { build, fake } from "@jackfranklin/test-data-bot";

interface Foo {
  value: boolean;
}

export const buildFoo = build<Foo>({
  fields: { value: fake((f) => f.random.boolean()) }
});
```

However, TypeScript compiler throws `ts(4023)` error, which in few words means that `buildFoo` type cannot be fully resolved because `BuildTimeConfig` type is not exported.

```txt
Exported variable 'buildFoo' has or is using name 'BuildTimeConfig' from external module
"/sandbox/node_modules/@jackfranklin/test-data-bot/build/index" but cannot be named. ts(4023)
```

In fact `buildFoo`  type is

```typescript
const buildFoo: (buildTimeConfig?: BuildTimeConfig<Foo> | undefined) => Foo;
```

and when comes the time to generate `.d.ts` file, TypeScript has no clue about what is the shape of `BuildTimeConfig`.

This issue does not apply to projects that have `declaration` flag not set in `tsconfig.json`, because no `.d.ts` file is generated.

Exporting `interface BuildTimeConfig` should be enough to fix, I have not tested it yet though.

[This example in codesandbox](https://codesandbox.io/s/xenodochial-feather-nk45f?file=/src/test-util.ts) shows the issue.

**Comment by chestercharles** [comment]
In addition to exporting the `BuildTimeConfig` interface to fix this issue, I think many more of the internal interfaces should be exported, such as `FieldsConfiguration`.

That would allow us to do something like this:
```
type Widget = { name: string };
const overridesForMale: FieldsConfiguration<Widget> = {
  name: fake(f => f.name.firstName(0))
}
const widget = build<Widget>('Widget', {
  fields: {
    name: fake(f => f.name.firstName())
  },
  traits: {
    male: {
      overrides: overridesForMale
    }
  }
}) ;
```

**Comment by JFigueroa51** [comment]
There's actually a PR that fixes this #344, but it's been open for some months. Is there anything stopping that merge?

**Comment by jackfranklin** [comment]
That PR has now been fixed, so v1.4.0 should solve this I think. Please open an issue if there's more that can be done here. Thanks!
